src/main/java/calibration/CalibrationApp.java

```diff
@@ -194,18 +194,21 @@ public void setPersistedFile(Optional<File> file) {
      * Loads person data from the specified file. The current data will be replaced.
      *
      * @param file
+     * @param headlessCompute true if computing headless, false if computing from the UI
      */
-    public void loadAnalysisFromFile(File file) {
+    public void loadAnalysisFromFile(File file, boolean headlessCompute) {
```

Comment:
`boolean` flag args like this are a code smell. They result in a difficult to understand API and there is usually a better way to handle things. A headless compute should not be calling a UI class `CalibrationApp`. Some reorganization might be neccesary to achieve the desired result.